pkg/auth/auth_test.go

```diff
@@ -17,68 +17,125 @@ limitations under the License.
 package auth
 
 import (
-	"strings"
+	"reflect"
 	"testing"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	discovery "k8s.io/client-go/discovery"
 	fakediscovery "k8s.io/client-go/discovery/fake"
 	"k8s.io/client-go/kubernetes/fake"
 )
 
-func TestCanI(t *testing.T) {
-	resourceList := metav1.APIResourceList{
+type fakeK8sAuth struct {
+	DiscoveryCli discovery.DiscoveryInterface
+}
+
+func (u fakeK8sAuth) Validate() error {
+	return nil
+}
+func (u fakeK8sAuth) GetResourceList(groupVersion string) (*metav1.APIResourceList, error) {
+	return u.DiscoveryCli.ServerResourcesForGroupVersion(groupVersion)
+}
+func (u fakeK8sAuth) CanI(verb, group, resource, namespace string) (bool, error) {
```

Comment:
this should be `GetForbiddenActions`, the signature is also different.

Reply:
This is the method of the internal `k8sInterface`, I didn't change the name of this method because this one really mimics the `can-i` command of `kubectl`

Reply:
Ok, sorry I'm not sure why I keep getting confused over this PR!